File: servers/typescript/src/id.ts
import { randomBytes } from "node:crypto";

export class IDGenerator {
  private counter = 0;

  clientId(): string {
    return "client_" + randomBytes(4).toString("hex");
  }

  resumeToken(): string {
    return "rt_" + randomBytes(8).toString("hex");
  }

  messageId(): string {
    this.counter++;
    return `srv_${this.counter}`;
  }
}

File: servers/python/src/starfish_server/id.py
from __future__ import annotations

import secrets


class IDGenerator:
    def __init__(self) -> None:
        self._counter = 0

    def client_id(self) -> str:
        return "client_" + secrets.token_hex(4)

    def resume_token(self) -> str:
        return "rt_" + secrets.token_hex(8)

    def message_id(self) -> str:
        self._counter += 1
        return f"srv_{self._counter}"
